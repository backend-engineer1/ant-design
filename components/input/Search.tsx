import React from 'react';
import classNames from 'classnames';
import Input, { InputProps } from './Input';
import Icon from '../icon';

export interface SearchProps extends InputProps {
  onSearch?: (value: string) => any;
}

export default class Search extends React.Component<SearchProps, any> {
  static defaultProps = {
    prefixCls: 'ant-input-search',
  };
  input: any;
  onSearch = () => {
    const { onSearch } = this.props;
    if (onSearch) {
      onSearch(this.input.refs.input.value);
    }
    this.input.focus();
  }
  render() {
    const { className, prefixCls, ...others } = this.props;
    delete (others as any).onSearch;
    const searchSuffix = (
      <Icon
        className={`${prefixCls}-icon`}
        onClick={this.onSearch}
        type="search"
      />
    );
    return (
      <Input
        onPressEnter={this.onSearch}
        {...others}
        className={classNames(prefixCls, className)}
        suffix={searchSuffix}
        ref={node => this.input = node}
      />
    );
  }
}
