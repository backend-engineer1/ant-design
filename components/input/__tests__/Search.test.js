import React from 'react';
import { mount } from 'enzyme';
import Search from '../Search';
import Button from '../../button';
import focusTest from '../../../tests/shared/focusTest';

describe('Input.Search', () => {
  focusTest(Search);

  it('should support custom button', () => {
    const wrapper = mount(
      <Search enterButton={<button>ok</button>} />
    );
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('should support custom Button', () => {
    const wrapper = mount(
      <Search enterButton={<Button>ok</Button>} />
    );
    expect(wrapper.render()).toMatchSnapshot();
  });
});
