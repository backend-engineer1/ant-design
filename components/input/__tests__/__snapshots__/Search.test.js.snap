// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input.Search should support custom Button 1`] = `
<Search
  enterButton={
    <Button
      ghost={false}
      loading={false}
      prefixCls="ant-btn"
    >
      ok
    </Button>
  }
  inputPrefixCls="ant-input"
  prefixCls="ant-input-search"
>
  <Input
    className="ant-input-search ant-input-search-enter-button"
    disabled={false}
    onPressEnter={[Function]}
    prefixCls="ant-input"
    suffix={
      <Button
        className="ant-input-search-button"
        ghost={false}
        loading={false}
        onClick={[Function]}
        prefixCls="ant-btn"
        size={undefined}
        type="primary"
      >
        <Button
          ghost={false}
          loading={false}
          prefixCls="ant-btn"
        >
          ok
        </Button>
      </Button>
    }
    type="text"
  >
    <span
      className="ant-input-search ant-input-search-enter-button ant-input-affix-wrapper"
    >
      <input
        className="ant-input"
        disabled={false}
        onKeyDown={[Function]}
        style={null}
        type="text"
      />
      <span
        className="ant-input-suffix"
      >
        <Button
          className="ant-input-search-button"
          ghost={false}
          key="enterButton"
          loading={false}
          onClick={[Function]}
          prefixCls="ant-btn"
          type="primary"
        >
          <button
            className="ant-btn ant-input-search-button ant-btn-primary"
            onClick={[Function]}
            type="button"
          >
            <Button
              ghost={false}
              key=".0"
              loading={false}
              prefixCls="ant-btn"
            >
              <button
                className="ant-btn"
                onClick={[Function]}
                type="button"
              >
                <span
                  key=".0"
                >
                  ok
                </span>
              </button>
            </Button>
          </button>
        </Button>
      </span>
    </span>
  </Input>
</Search>
`;

exports[`Input.Search should support custom button 1`] = `
<Search
  enterButton={
    <button>
      ok
    </button>
  }
  inputPrefixCls="ant-input"
  prefixCls="ant-input-search"
>
  <Input
    className="ant-input-search ant-input-search-enter-button"
    disabled={false}
    onPressEnter={[Function]}
    prefixCls="ant-input"
    suffix={
      <button
        onClick={[Function]}
      >
        ok
      </button>
    }
    type="text"
  >
    <span
      className="ant-input-search ant-input-search-enter-button ant-input-affix-wrapper"
    >
      <input
        className="ant-input"
        disabled={false}
        onKeyDown={[Function]}
        style={null}
        type="text"
      />
      <span
        className="ant-input-suffix"
      >
        <button
          onClick={[Function]}
        >
          ok
        </button>
      </span>
    </span>
  </Input>
</Search>
`;
